pkg/autodiscovery/listeners/ecs.go

```diff
@@ -104,7 +104,13 @@ func (l *ECSListener) Stop() {
 // compares the container list to the local cache and sends new/dead services
 // over newService and delService accordingly
 func (l *ECSListener) refreshServices(firstRun bool) {
-	meta, err := ecsmeta.V2().GetTask()
+	client, err := ecsmeta.V2()
+	if err != nil {
+		log.Error(err)
```

Comment:
Let's use a less severe log level than `log.Error`. Typically `log.Debug` at most should be used if the only case that can be logged is that the metadata provider was disabled through configuration.

(also applies to all the other `log.Error` in this PR I believe)

Reply:
So `log.Error` should be used if we can expect a multitude of potentially vague errors?

Reply:
I think we shouldn't use `log.Error` for anything that'll be logged repeatedly, unless it's a serious error that affects crucial features of the Agent.

If multiple types of errors can happen -- each with their own severity that should be logged with different severities as well -- one approach is to create types for these errors, check for these error types in the caller, and log accordingly.